torch/_inductor/codegen/cpp.py

```diff
@@ -4964,11 +4964,19 @@ def is_contiguous_index(x):
                         ):
                             continue
                         # Local Buffer is a view of global buffer
+                        local_buffer_stride: list[int] = []
+                        stride = global_buffer_layout.stride[-1]
+                        local_buffer_size = get_call_ranges(scheduler_node)[
```

Comment:
is this case `scheduler_node` also a view of `global_buffer`?

Reply:
In this case `global_buffer` is `scheduler_node.node`.

Reply:
Because `global_buffer_layout` is the tensor layout while size_offset is the loop depth. The dimension of `global_buffer_layout` may not be the same as loop number, e.g, there are merged dims.
In this case, `global_buffer_layout` size is [5, 1, 32, 32] but the callrange is [5, 1024]. If use  `global_buffer_layout.size[size_offset:]` to create `local_buffer_layout` we get the size [32] but we need [1024].